lib/src/widgets/typing_indicator.dart

```diff
@@ -0,0 +1,487 @@
+                    ),
+                  )
+                : const SizedBox(),
+          ],
+        ),
+      );
+
+  /// Handler for circles offset.
+  Animation<Offset> _circleOffset(
+    Offset? start,
+    Offset? end,
+    Interval animationInterval,
+  ) =>
+      TweenSequence<Offset>(
+        <TweenSequenceItem<Offset>>[
+          TweenSequenceItem<Offset>(
+            tween: Tween<Offset>(
+              begin: start,
+              end: end,
+            ),
+            weight: 50.0,
+          ),
+          TweenSequenceItem<Offset>(
+            tween: Tween<Offset>(
+              begin: end,
+              end: start,
+            ),
+            weight: 50.0,
+          ),
+        ],
+      ).animate(CurvedAnimation(
+        parent: _animatedCirclesController,
+        curve: animationInterval,
+        reverseCurve: animationInterval,
+      ));
+}
+
+/// Typing Widget.
+class TypingWidget extends StatelessWidget {
+  const TypingWidget({
+    super.key,
+    required this.widget,
+    required this.context,
+    required this.mode,
+  });
+
+  final TypingIndicator widget;
+  final BuildContext context;
+  final TypingIndicatorMode mode;
+
+  @override
+  Widget build(BuildContext context) {
+    final sWidth = _getStackingWidth(
+      widget.options.typingUsers,
+      MediaQuery.of(context).size.width,
+    );
+    if (mode == TypingIndicatorMode.text) {
+      return SizedBox(
+        child: Text(
+          _multiUserTextBuilder(widget.options.typingUsers),
+          style: InheritedChatTheme.of(context)
+              .theme
+              .typingIndicatorTheme
+              .multipleUserTextStyle,
+        ),
+      );
+    } else if (mode == TypingIndicatorMode.avatar) {
+      return SizedBox(
+        width: sWidth,
+        child: AvatarHandler(
+          context: context,
+          author: widget.options.typingUsers,
+        ),
+      );
+    } else {
+      return Row(
+        children: <Widget>[
+          SizedBox(
+            width: sWidth,
+            child: AvatarHandler(
+              context: context,
+              author: widget.options.typingUsers,
+            ),
+          ),
+          const SizedBox(width: 10),
+          Text(
+            _multiUserTextBuilder(widget.options.typingUsers),
+            style: InheritedChatTheme.of(context)
+                .theme
+                .typingIndicatorTheme
+                .multipleUserTextStyle,
+          ),
+        ],
+      );
+    }
+  }
+
+  /// Handler for multi user typing text.
+  String _multiUserTextBuilder(List<types.User> author) {
+    if (author.isEmpty) {
+      return '';
+    } else if (author.length == 1) {
+      return '${author.first.firstName} is typing';
+    } else if (author.length == 2) {
+      return '${author.first.firstName} and ${author[1].firstName}';
+    } else {
+      return '${author.first.firstName} and ${author.length - 1} others';
+    }
+  }
```

Comment:
how will this fly with i18n?

Reply:
The localization part will be handled by package owners, as discussed.